File: Lodjinha/app/src/main/java/br/com/bsavoini/lodjinha/home/BannerPagerAdapter.java
package br.com.bsavoini.lodjinha.home;

import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import br.com.bsavoini.lodjinha.R;
import br.com.bsavoini.lodjinha.api.model.BannerModel;
import com.squareup.picasso.Picasso;

import java.util.List;

public class BannerPagerAdapter extends PagerAdapter {
    private List<BannerModel> banners;

    public BannerPagerAdapter(List<BannerModel> banners) {
        this.banners = banners;
    }

    @Override
    public int getCount() {
        return banners.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object obj) {
        return obj == view;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        LayoutInflater inflater = LayoutInflater.from(container.getContext());
        View view = inflater.inflate(R.layout.item_banner, container, false);
        container.addView(view);

        ImageView imgBanner = view.findViewById(R.id.img_banner);

        Picasso.with(container.getContext())
                .load(banners.get(position).getImageURL())
                .placeholder(R.drawable.menu_pattern)
                .into(imgBanner);

        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((View) object);
    }

}

File: Lodjinha/app/src/main/kotlin/com/b2w/lodjinha/ui/home/banner/BannerPagerAdapter.kt
package com.b2w.lodjinha.ui.home.banner

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.support.v4.app.FragmentActivity
import android.support.v4.view.PagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.b2w.lodjinha.R
import com.b2w.lodjinha.data.model.BannerItem
import com.squareup.picasso.Picasso
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity

class BannerPagerAdapter(val banners: List<BannerItem>): PagerAdapter(){

    override fun isViewFromObject(p0: View, p1: Any): Boolean {
        return p0.equals(p1)
    }

    override fun getCount(): Int {
        return banners.size
    }

    override fun destroyItem(container: ViewGroup, position: Int, obj: Any) {
        container.removeView(obj as View)
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val imageLayout = LayoutInflater.from(container.context).inflate(R.layout.view_banner_item, container, false)
        val imageView : ImageView = imageLayout.findViewById(R.id.view_banner_item_img)

        Picasso.get()
            .load(banners[position].image)
            .into(imageView)

        (container as ViewPager).addView(imageLayout)

        imageLayout.setOnClickListener{
            val intent = Intent(Intent.ACTION_VIEW)
            intent.data = Uri.parse(banners[position].link)
            (container.context as FragmentActivity).startActivity(intent)
        }

        return imageLayout
    }
}